### ISSUE TITLE
whitenoise & django-compressor

### ISSUE THREAD

**Issue by racitup** [awesto/django-shop#645]
The framework supplies and example shop uses a sekizai processor that uses django-compressor to optimise js and css files. Due to [django-compressor issue 351](https://github.com/django-compressor/django-compressor/issues/351) it appears this can't be used offline.
When trying to use whitenoise to serve static files it appears there is an incompatibility between the two. Is there any experience or advice of using both together or must django-compressor be removed?

**Comment by racitup** [comment]
For others with the same problem it seems that the django-compressor integration and whitenoise are incompatible and are different solutions for solving an overlapping problem.

The incompatibility is that whitenoise will only serve files that are in its manifest file which is generated by the `collectstatic` command. It seems to do this by pulling in all the files in all the static directories.

The django-compressor integration generates new files on the fly in a `CACHE` directory when a page is rendered by combining existing static files. There seems to be no way of generating these files beforehand. I guess this could be a problem in a production environment where the django app only has read permission on the static file directory.

So there seems to be two ways of solving it:
1. remove the django-compressor integration and use whitenoise solely
2. use the django-compressor integration and use [dj-static/Cling](https://github.com/kennethreitz/dj-static)

I think there are pros and cons to each. The main pro for whitenoise I can see is its header optimisations for caching.
Anyone trying to deploy django-SHOP on Heroku will run into this problem. I've picked number 2 and will update here if I find any other related issues. The reason I picked 2 is the combined static files seems like a neat solution, and dj-static also comes with the ability to serve media, which is another problem to solve.
